### ISSUE TITLE
Ensure open files do not accumulate

### ISSUE THREAD

**Issue by dginev** [dginev/CorTeX#33]
After ~130,000 jobs the dispatcher became unresponsive with:
```
Incoming sink job 130340 for service "tex_to_html", taskid: 1588123
-- Error TODO: File::create(recv_path): Os { code: 24, kind: Other, message: "Too many open files" }
Incoming sink job 130341 for service "_unknown_", taskid: -1
```

It looks like we need to reap a lot more of the open files, and in particular (I assume) the zmq connections that are left dangling. These can accumulate when there are distantly remote machines that drop connection often - such as the Erlangen-NYC distance for my home working machine.

**Comment by dginev** [comment]
Log trace of that session:

[dispatcher.log.1.gz](https://github.com/dginev/CorTeX/files/2280277/dispatcher.log.1.gz)

**Comment by dginev** [comment]
A practical solution may be to also up the allowed maximum on the host machine, since it's a bit low at the moment:

```
$ ulimit -n
1024
```

will retry the current run with `ulimit -n 99999`

**Comment by dginev** [comment]
It seems the multiple NICs are the core reason here as described in:
https://zeromq.jira.com/browse/LIBZMQ-556

On the machines causing the large number of open files, I also see a significant number of reconnection problems, which is why I had resorted to *not* reusing identities, causing the open file accumulation.

Hm...

**Comment by dginev** [comment]
Fixed via 91daa23eb9b36c9f5e41f7a4ab592fa5a52d08be

Using the `ZMQ_ROUTER_HANDOVER` socket option allowed to reconnect from a different NIC when using the same identity, and combined with the "fixed identities" introduced in the latest 0.4 of LaTeXML-Plugin-Cortex, this acts robustly.

After 20,000 jobs by the dispatcher, the worker machines are still successfully reconnecting and processing new tasks. 

This is actually a re-fix for issue #20, and I will continue monitoring the open files as the conversion progresses, as I think there are still _some_ leaked file descriptors, monitoring `lsof -a -p dispatcher_pid`

**Comment by dginev** [comment]
Can confirm the router handover setting worked very robustly for this particular cause of instability, which is another sigh of relief.
